src/RedisOutputCachingMiddleware/RedisOutputCachingMiddleware.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Web.Redis;
+using StackExchange.Redis;
+
+namespace RedisOutputCachingMiddleware
+{
+    public class OutputCachingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IDatabase _cache;
+        // optional expiration time, default to 1 day if not defined 
+        private int _ttl = (int)TimeSpan.FromDays(1).TotalSeconds;
+        
+        public OutputCachingMiddleware(RequestDelegate next, string redisConnectionString, [Optional] int ttl)
+        {
+            _next = next;
+            try
+            {
+                _cache = ConnectionMultiplexer.Connect(redisConnectionString).GetDatabase();
+            }
+            catch (Exception ex)
+            {
+                LogUtility.LogError($"Cannot connect to Redis: {ex.Message}");
+            }
+            
+            _ttl = ttl;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            // use the url, header, and request body as a key 
+            string pathAndQuery = context.Request.GetEncodedPathAndQuery();
+            string headers = context.Request.Headers.ToString();
+            string body = context.Request.Body.ToString();
+            RedisKey key = $"{pathAndQuery}{headers}{body}";
```

Comment:
Preceding lines can be condensed to a single line like: 
`RedisKey key = $"{context.Request.GetEncodedPathAndQuery()}{context.Request.Headers}{context.Request.Body}";`